File: SemanticSearch/Node.js/Web/Plain/src/cache/adapters/Local.js
class Local {
  constructor() {
    this._store = new Map();
  }

  get(key) {
    const entry = this._store.get(key);
    if (!entry) return null;
    if (entry.exp < Date.now()) {
      this._store.delete(key);
      return null;
    }
    return entry.val;
  }

  set(key, value, ttl = 300) {
    this._store.set(key, { val: value, exp: Date.now() + ttl * 1000 });
  }

  del(key) {
    this._store.delete(key);
  }
}

module.exports = Local;

File: DataPipeline/Python/Web/Plain/src/cache/adapters/local.py
import time
from cache.cache_adapter import CacheAdapter


class Local(CacheAdapter):
    def __init__(self):
        self._store = {}

    def get(self, key):
        entry = self._store.get(key)
        if entry is None:
            return None
        if entry["exp"] < time.time():
            del self._store[key]
            return None
        return entry["val"]

    def set(self, key, value, ttl=300):
        self._store[key] = {"val": value, "exp": time.time() + ttl}

    def delete(self, key):
        self._store.pop(key, None)